### PR TITLE
refactor: Working on secrets-store-csi-driver-provider-aws

### PR DESCRIPTION
**Pull request by fcarta29**
### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/aws-ia/terraform-aws-eks-blueprints-addons/blob/main/.github/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

Refactoring secrets-store-csi-driver-provider-aws for v5

### Motivation

- Resolves #<issue-number>

### More

- [ ] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [X] Yes, I ran `pre-commit run -a` with this PR

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: locals.tf [hunk 1]
```diff
@@ -43,6 +43,11 @@ locals {
       enable             = true
       serviceAccountName = local.secrets_store_csi_driver_service_account
     } : null
+    csiSecretsStoreProviderAws = var.enable_csi_secrets_store_provider_aws && var.enable_csi_secrets_store_provider_aws_gitops ? {
+      enable             = true
+      serviceAccountName = local.csi_secrets_store_provider_aws_service_account
+    } : null
```
Comment (askulkarni2) [inline locals.tf:49]: The serviceAccountName is not needed to be passed. The service account needs to be IRSA annotated for the workload consuming the secret not the csi driver here.

```suggestion
    csiSecretsStoreProviderAws = var.enable_csi_secrets_store_provider_aws && var.enable_csi_secrets_store_provider_aws_gitops ? {
      enable = true
    } : null
```

File: main.tf [hunk 2]
```diff
@@ -2052,6 +2052,66 @@ module "secrets_store_csi_driver" {
   tags = var.tags
 }
 
+################################################################################
+# CSI Secrets Store Provider AWS
+################################################################################
+
+locals {
+  csi_secrets_store_provider_aws_name            = "secrets-store-csi-driver-provider-aws"
+  csi_secrets_store_provider_aws_service_account = try(var.csi_secrets_store_provider_aws_name.service_account_name, "${local.csi_secrets_store_provider_aws_name}-sa")
```
Comment (askulkarni2) [inline main.tf:2061]:
```suggestion
  csi_secrets_store_provider_aws_name  = "secrets-store-csi-driver-provider-aws"
```

File: main.tf [hunk 3]
```diff
@@ -2052,6 +2052,66 @@ module "secrets_store_csi_driver" {
...
+}
+
+module "csi_secrets_store_provider_aws" {
+
+  # source = "aws-ia/eks-blueprints-addon/aws"
+  source = "./modules/eks-blueprints-addon"
+
+  create = var.enable_secrets_store_csi_driver_provider_aws
+
+  # https://github.com/aws/eks-charts/blob/master/stable/csi-secrets-store-provider-aws/Chart.yaml
```
Comment (askulkarni2) [inline main.tf:2071]:
```suggestion
  # https://github.com/aws/secrets-store-csi-driver-provider-aws/blob/main/charts/secrets-store-csi-driver-provider-aws/Chart.yaml
```

File: main.tf [hunk 4]
```diff
@@ -2052,6 +2052,66 @@ module "secrets_store_csi_driver" {
...
+  name             = try(var.csi_secrets_store_provider_aws.name, local.csi_secrets_store_provider_aws_name)
+  description      = try(var.csi_secrets_store_provider_aws.description, "A Helm chart to install the Secrets Store CSI Driver and the AWS Key Management Service Provider inside a Kubernetes cluster.")
```
Comment (askulkarni2) [inline main.tf:2073]:
```suggestion
  description      = try(var.csi_secrets_store_provider_aws.description, "A Helm chart for the AWS Secrets Manager and Config Provider for Secret Store CSI Driver.")
```

File: variables.tf [hunk 5]
```diff
@@ -645,17 +667,35 @@ variable "velero_backup_s3_bucket" {
...
+  description = "Enable Promtail add-on"
+  type        = bool
+  default     = false
+}
+
+variable "promtail_helm_config" {
+  description = "Promtail Helm Chart config"
+  type        = any
+  default     = {}
 }
```
Comment (askulkarni2) [inline variables.tf:699]: These addons are removed in main.

**Review by askulkarni2 — COMMENTED** [review]
Also missing updates to `locals.tf`

**Comment by fcarta29** [comment]
Closing this PR to open a new one without conflicts

**Pull request closed without merging**